File: core/src/jvmMain/java/de/haukesomm/sokoban/core/moving/MoveCoordinatorResult.java
package de.haukesomm.sokoban.core.moving;

import de.haukesomm.sokoban.core.GameState;
import de.haukesomm.sokoban.core.moving.validation.MoveValidatorStatus;

import java.util.Optional;
import java.util.Set;

public record MoveCoordinatorResult(
        boolean success,
        Set<MoveValidatorStatus> moveValidatorStatuses,
        Optional<GameState> gameState
) {
}

File: core/src/commonMain/kotlin/de/haukesomm/sokoban/core/moving/MoveCoordinatorResult.kt
package de.haukesomm.sokoban.core.moving

import de.haukesomm.sokoban.core.state.GameState
import de.haukesomm.sokoban.core.moving.validation.MoveValidatorStatus

data class MoveCoordinatorResult(
    val success: Boolean,
    val moveValidatorStatuses: Set<MoveValidatorStatus>,
    val gameState: GameState?
)
